Add AllowlistSynchronizer documentation to falcon-sensor (#405)
CrowdStrike/falcon-helm: helm-charts/falcon-sensor/README.md

helm show values crowdstrike/falcon-sensor
```

> ### GKE Autopilot Configuration

Author: @cs-alphonse Updated.

> Due to the Daemonset pods being run as a privileged, the following additional manifest is required to be applied to the cluster before deployment:

Reviewer: ```suggestion
Due to the Daemonset pods being run in privileged mode, the following additional manifest is required to be applied to the cluster before deployment:
```

```
apiVersion: auto.gke.io/v1
kind: AllowlistSynchronizer
metadata:
  name: crowdstrike-synchronizer
spec:
  allowlistPaths:

>   - CrowdStrike/falcon-sensor/*

Reviewer: It looks like Alphonse provided more specific paths in the other PR https://github.com/CrowdStrike/falcon-helm/pull/393/files#r2061982966
```
- CrowdStrike/falcon-sensor/cleanup/crowdstrike-falconsensor-cleanup-allowlist-v1.0.1.yaml
- CrowdStrike/falcon-sensor/deploy/crowdstrike-falconsensor-deploy-allowlist-v1.0.0.yaml
- CrowdStrike/falcon-sensor/deploy/crowdstrike-falconsensor-deploy-allowlist-v1.0.2.yaml
- CrowdStrike/falcon-sensor/falconctl/crowdstrike-falconsensor-falconctl-allowlist-v1.0.1.yaml
```

Author: @mr-jungchoi Let's do the wildcard for now. Once we start getting more allowlists we can pin a version or wildcard for version.  It's not really an issue now since the webhook only needs to check matches for a few templates.

helm show values crowdstrike/falcon-sensor
```

### GKE Autopilot Configuration
#### Configuring the AllowlistSynchronizer

> Running Daemonset Pods with privileged access on GKE Autopilot requires special configurations due to default security restrictions. To enable these privileged Pods, you need to implement an AllowlistSynchronizer. This resource applies WorkloadAllowlists to your cluster, which the GKE Autopilot validating webhook uses to approve Pod deployments based on their manifest spec and image digests. Follow these steps to properly configure the AllowlistSynchronizer:

Reviewer 2: ```suggestion
Running Daemonset Pods with privileged access on GKE Autopilot requires special configurations due to default security restrictions. To enable these privileged Pods, you need to configure an AllowlistSynchronizer. This resource applies CrowdStrike specific WorkloadAllowlists to your cluster, which the GKE Autopilot validating webhook uses to approve Pod deployments based on their manifest spec and image digests. Follow these steps to properly configure the AllowlistSynchronizer:
Comment

```

```
##### WorkloadAllowlist Definitions
The WorkloadAllowlists serve the following purposes:
- crowdstrike-falconsensor-cleanup-allowlist-vX.X.X: Authorizes the Falcon Sensor Cleanup DaemonSet to operate within the cluster.
- crowdstrike-falconsensor-deploy-allowlist-vX.X.X: Permits the deployment and execution of the Falcon Sensor Deploy DaemonSet in the cluster environment.
- crowdstrike-falconsensor-falconctl-allowlist-vX.X.X: Enables the Falconctl job to run, facilitating sensor configuration and management tasks.

> [!NOTE]
> Additional information about AllowlistSynchronizer can be found here: [https://cloud.google.com/kubernetes-engine/docs/reference/crds/allowlistsynchronizer](https://cloud.google.com/kubernetes-engine/docs/reference/crds/allowlistsynchronizer)

#### Obtaining an Authorized Image
WorkloadAllowlists ensure that only authorized container images are deployed to pods by verifying their image digests. To view the list of approved image digests, execute the following command:
```
kubectl get workloadallowlists <crowdstrike-falconsensor-XXXXXXX-allowlist-vX.X.X>  -o=jsonpath='{range .containerImageDigests[*].imageDigests[*]}{@}{"\n"}{end}'
```

> To obtain the Falcon Container sensor image, you have two options:

Reviewer 2: ```suggestion
To obtain the Falcon Node sensor image, you have two options:
```

1. Pull directly from the CrowdStrike registry
2. Copy the image from the CrowdStrike registry to your private registry

For option 2, we provide an automation script to simplify the process:
[https://github.com/CrowdStrike/falcon-scripts/tree/main/bash/containers/falcon-container-sensor-pull](https://github.com/CrowdStrike/falcon-scripts/tree/main/bash/containers/falcon-container-sensor-pull)

> When copying images to a private registry, it's crucial to preserve the image digest. We recommend using tools like Skopeo for this purpose, as they ensure the integrity of the image during transfer.

Reviewer 2: ```suggestion
When copying images to a private registry, it's crucial to preserve the image digest. We recommend using tools like Skopeo for this purpose, as they ensure the digest of the image remains the same after the transfer.
```
